src/parser/mod.rs

```diff
@@ -266,6 +266,22 @@ impl ParserOptions {
         self.unescape = unescape;
         self
     }
+
+    /// Set if semicolon statement delimiters are required.
```

Comment:
What is the usecase to have the same configuration information on the `ParserOptions` and the `Dialect`?

It seems there are a few places in the Parser that inconsistently check the options and/or the dialect flag. If we had only one way to specify the option, there would not be any room for inconsistencies

 Or maybe I misunderstand what is going on here

Reply:
> What is the usecase to have the same configuration information on the ParserOptions and the Dialect?

It's similar to the existing `with_trailing_commas` logic, and has several practical use cases:

1. can be set to `true` for SQL Server & left `false` for all others (until if/when someone discovers this is supported in other dialects)
2. integration with existing testing patterns. eg, we can have `all_dialects_requiring_semicolon_statement_delimiter` & `all_dialects_not_requiring_semicolon_statement_delimiter`
3. used to set the corresponding default parser option

It seems like testing parsing without semicolons would get a lot worse if it _wasn't_ a dialect option

Reply:
> In terms of which way to configure, I'm guessing it could make more sense to have the flag as a parser setting

This seems reasonable to me in terms of philosophy. However, I don't know how that could be implemented practically. I.e., how would you write that code in this project to be able to have the same test coverage?

The benefit of setting the parse option based on the dialect configuration is:

1. not all dialects support parsing without semicolons (so far, only one)
2. for dialects that require semicolons, we probably don't want anything to change (as is implemented here in the PR)
3. this project's tests are oriented around dialect features, so implementing the semicolon requirement as a dialect feature makes it more clear for how to run the tests